roles/repository/tasks/main.yml

```diff
@@ -18,11 +18,11 @@
       ansible.builtin.set_fact:
         temp_array: []
 
-    - name: Set solr.sharedSecret property
+    - name: Set solr.sharedSecret property when search is enabled
       ansible.builtin.set_fact:
         temp_array: >-
           {{ temp_array + [ '-Dsolr.sharedSecret=' ~ reposearch_shared_secret | quote ] }}
-      when: acs.version is version('7.2.0', '>=')
+      when: (acs.version is version('7.2.0', '>=')) and ((groups.search | default([])) | length > 0)
```

Comment:
it works, but you can't really have logic related to the inventory inside a role

role should just handle the presence or not of this secret (eventually making it an explicit role argument)

the logic should be in the parent playbook where we have put most of this kind of logic - now I am probably remembering why we created a separate task for this